core/src/main/java/io/parsingdata/metal/expression/value/Elvis.java

```diff
@@ -32,25 +35,27 @@
  * evaluates to that value, else it evaluates to the value of <code>ref("bar")</code>.
  */
 public class Elvis implements ValueExpression {
-    private final ValueExpression _lop;
-    private final ValueExpression _rop;
+    private final ValueExpression lop;
+    private final ValueExpression rop;
 
     public Elvis(final ValueExpression lop, final ValueExpression rop) {
-        _lop = lop;
-        _rop = rop;
+        this.lop = checkNotNull(lop, "lop");
+        this.rop = checkNotNull(rop, "rop");
     }
 
     @Override
-    public OptionalValue eval(final Environment env, final Encoding enc) {
-        final OptionalValue eval = _lop.eval(env, enc);
-        if (eval.isPresent()) {
-            return eval;
-        }
-        return _rop.eval(env, enc);
+    public OptionalValueList eval(final Environment env, final Encoding enc) {
+        return eval(lop.eval(env, enc), rop.eval(env, enc));
+    }
+
+    private OptionalValueList eval(final OptionalValueList llist, final OptionalValueList rlist) {
+        if (llist.isEmpty()) { return rlist; }
+        if (rlist.isEmpty()) { return llist; }
```

Comment:
My earlier comment holds here as well, these two lines constitute explicit behavior that should be properly documented.